File: scripts/generate-icons.js
#!/usr/bin/env node
/**
 * Generate simple PNG icons for PWA
 * Creates minimal valid PNG files with StudyFlow branding
 * Uses base64-encoded minimal PNG structure
 */

const fs = require('fs')
const path = require('path')

const ICON_DIR = 'public/icons'
const PRIMARY_COLOR = '#4F46E5' // Indigo from theme

// Minimal valid PNG structure (1x1 pixel, transparent)
// We'll create a simple colored square
function createMinimalPNG(size, color) {
  // PNG signature
  const pngSignature = Buffer.from([0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A])
  
  // For simplicity, we'll create a base64-encoded minimal PNG
  // This is a 1x1 red pixel PNG (we'll scale it conceptually)
  // In production, you should use proper image generation library
  
  // For now, create a simple approach: use a data URL approach
  // But since we need actual files, let's create a minimal valid PNG
  
  // Minimal PNG structure (simplified - in production use sharp or canvas)
  // This creates a basic PNG that browsers will accept
  const minimalPNG = Buffer.from(
    'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg==',
    'base64'
  )
  
  return minimalPNG
}

// Better approach: Create a simple SVG and convert, or use a library
// For now, let's create a script that uses ImageMagick or provides instructions

function generateIcons() {
  console.log('🎨 Generating PWA icons...')
  
  // Ensure directory exists
  if (!fs.existsSync(ICON_DIR)) {
    fs.mkdirSync(ICON_DIR, { recursive: true })
  }
  
  // Check if we can use a simpler approach
  // Create a simple colored square PNG using a minimal valid structure
  // Note: This is a workaround - proper solution needs image library
  
  console.log('⚠️  This script requires an image library.')
  console.log('📝 Options:')
  console.log('   1. Install sharp: npm install --save-dev sharp')
  console.log('   2. Use ImageMagick: brew install imagemagick')
  console.log('   3. Use online tool: https://realfavicongenerator.net/')
  console.log('')
  console.log('💡 Quick fix: Create simple colored squares manually')
  console.log('   - Use any image editor')
  console.log('   - Create 192x192px square with color #4F46E5')
  console.log('   - Save as PNG')
  console.log('   - Repeat for 512x512px')
  
  // For now, we'll create a placeholder that indicates the need for real icons
  const placeholder = Buffer.from(
    'iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg==',
    'base64'
  )
  
  // Write minimal valid PNG (1x1 transparent)
  // This will prevent the "invalid image" error, but icons won't look good
  fs.writeFileSync(path.join(ICON_DIR, 'icon-192x192.png'), placeholder)
  fs.writeFileSync(path.join(ICON_DIR, 'icon-512x512.png'), placeholder)
  
  console.log('✅ Created minimal placeholder icons (1x1 transparent)')
  console.log('⚠️  Please replace with proper 192x192 and 512x512 icons!')
}

if (require.main === module) {
  generateIcons()
}

module.exports = { generateIcons }

File: scripts/generate-icons.py
#!/usr/bin/env python3
"""
Generate simple PNG icons for PWA
Creates valid PNG files with StudyFlow branding
"""
from PIL import Image, ImageDraw, ImageFont
import os

ICON_DIR = "public/icons"
PRIMARY_COLOR = "#4F46E5"  # Indigo from theme
TEXT_COLOR = "#FFFFFF"  # White

def create_icon(size, filename):
    """Create a square icon with StudyFlow branding"""
    # Create image with primary color background
    img = Image.new('RGB', (size, size), PRIMARY_COLOR)
    draw = ImageDraw.Draw(img)
    
    # Try to use a nice font, fallback to default
    try:
        # Try to use system font
        font_size = int(size * 0.3)
        font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", font_size)
    except:
        try:
            font = ImageFont.truetype("/usr/share/fonts/truetype/dejavu/DejaVuSans-Bold.ttf", int(size * 0.3))
        except:
            font = ImageFont.load_default()
    
    # Draw "SF" text in center
    text = "SF"
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    
    position = ((size - text_width) // 2, (size - text_height) // 2)
    draw.text(position, text, fill=TEXT_COLOR, font=font)
    
    # Save icon
    os.makedirs(ICON_DIR, exist_ok=True)
    img.save(filename, 'PNG')
    print(f"✅ Created {filename} ({size}x{size})")

if __name__ == "__main__":
    print("🎨 Generating PWA icons with Python...")
    create_icon(192, f"{ICON_DIR}/icon-192x192.png")
    create_icon(512, f"{ICON_DIR}/icon-512x512.png")
    print("✅ All icons generated!")

